dev/tests/integration/framework/Magento/TestFramework/Db/DynamicTables.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\TestFramework\Db;
+
+use Magento\TestFramework\Db\DymanicTables\CategoryProductIndexTables;
+
+/**
+ * Class to pre-create dynamic tables
+ */
+class DynamicTables
+{
+    /**
+     * @var CategoryProductIndexTables
+     */
+    private $categoryProductIndexTables;
+
+    /**
+     * @param CategoryProductIndexTables $categoryProductIndexTables
+     */
+    public function __construct(
+        CategoryProductIndexTables $categoryProductIndexTables
+    ) {
+        $this->categoryProductIndexTables = $categoryProductIndexTables;
+    }
+
+    /**
+     * Create dynamic tables before the test to preserve integration tests isolation
+     */
+    public function createTables()
+    {
+        $this->categoryProductIndexTables->createTables();
```

Comment:
@akaplya Do we have task to handle this in more abstract way so theses changes can be merged to 2.4-develop?